docs/userguide/common-patterns.md

```diff
@@ -34,15 +34,59 @@ simply by rendering various children. It may just combine the renderings of mult
 use its props to determine which of a set of children to render. Such workflows can often be
 stateless.
 
-## One-and-done Workflows (RenderingT v. OutputT)
+## Skippable Workflows
 
-A common question is “why can’t I emit output from `initialState`,” or “what if my Workflow realizes it doesn’t actually need to run? The most efficient, and most expressive, way to handle this is to use an optional or conditional `Rendering` type, and an `Output` of [`Never`](https://nshipster.com/never/)/[`Nothing`](https://medium.com/@agrawalsuneet/the-nothing-type-kotlin-2e7df43b0111).
+* Why can’t I emit output from `initialState`?
+* Why can’t I emit output from `render`?
+* Why can't I change states during `render`?
+* What if while rendering my Workflow realizes it doesn’t actually need to run?
+
+The best pattern we've found for all of these situations is a "skippable workflow".
+The idea is that a workflow that performs work that may not actually be necessary
+provides an extra function for parents to call.
+The parent can call this during the upstream action
+where it's deciding whether or not to enter the state
+in which the child would be run.
+
+Perhaps it's a simple boolean function indicating whether or not the child actually has any work to do:
+
+```kotlin
+state = if (stepTwoWorkflow.isThisNecessary()) RunStepTwo else RunStepThree
```

Comment:
This only works if the `stepTwoWorkflow` is created and stored in the `state` a head of time, before you get to the `apply(action)` method. If your `stepTwoWorkflow` can't be created until you get into the `apply(action)` with a specific action/state, then you are out of luck.

Reply:
Typically in Kotlin child workflows are DI injected (available as constructor parameters fo the parent), and actions are inner objects of the parent workflow with access to its injected dependencies.

In Swift, didn't a change just land to make the parent workflow available from the action? cc @jamieQ